### DISCUSSION TITLE
Error in observable for Eta LFV decays

### DISCUSSION THREAD

**Discussion by Joro20** [flav-io/flavio#260] [Q&A] [answered]
Hello,

I was trying to estimate some LFV decays from eta/eta^{\prime} meson and was using the code available in the Flavio, but it seems complaining about the observable itself! The MWE is below:

```python
import numpy as np
print('loading wilson and flavio')
from wilson import Wilson
import flavio
`print('done\n')
``` 

```python
import random
for i in np.arange(1,50,0.5):,
random2=random.uniform(-2.5,-2.3),
ref_WC2=10**(-random2),
ref_scale = 1e3  #in GeV units,
WC2 = 'lq3_1223' ,
wc2  = Wilson({WC2: ref_WC2/ref_scale/ref_scale}, scale=ref_scale, eft='SMEFT', basis='Warsaw'),
BR_Etab1Smue=flavio.np_prediction('BR(eta_b(1S)->mue)',wc2),
print(ref_WC2,BR_Etab1Smue)
``` `


But the error message is

Keyerror:'BR(eta_b(1S)->mue)'

I'm not sure what is wrong! Please help.

**Comment by DavidMStraub** [comment]
Please check the docs for observables that exist: https://flav-io.github.io/docs/observables.html

**Reply by Joro20** [reply]
It is indeed a listed listed observable under the quarkonium lepton decays! Please see the attachment.
<img width="910" alt="Screenshot 2024-10-24" src="https://github.com/user-attachments/assets/2a35feed-e25a-41a1-81f5-1a0ce8eb04ef">

**Comment by peterstangl** [comment]
I'm not able to reproduce the error you see using the latest flavio version.

Which version of flavio are you using? You can check e.g. using  `print(flavio.__version__)`. The LFV quarkonium decays have been added in version 2.5.0 (see https://github.com/flav-io/flavio/releases/tag/v2.5.0). If you use an older version of flavio, these observables are not available.

**Reply by Joro20** [reply]
Hi Peter,

Thanks for your suggestion, you might be right about the version! I'm using these on Windows 10 and after upgrading while I ran the command `pip list` in my cmd it showed that the version of `flavio` is 0.25! Is that correct ?

But while I tried to run the MWE it is complaining about the `wilson` module!
<img width="566" alt="Screenshot 2024-10-25_1" src="https://github.com/user-attachments/assets/e5c1542d-313a-4dce-b1cc-043e2d946ea5">
 I then tried to reinstall `wilson` and it encountered with the dependency issues (please see the attached images) 
<img width="856" alt="Screenshot 2024-10-25_2" src="https://github.com/user-attachments/assets/5f3f0c50-ee47-45f3-ab2e-170844469d2d">
! Is there any issue with the compatibility between the versions of Python (my version is 3.13.0), Flavio and Wilson ? I'm very much confused!

**Reply by DavidMStraub** [reply]
@peterstangl could this be due to PyRunDec not having Python 3.13 wheels yet?

**Reply by peterstangl** [reply]
@DavidMStraub, yes it seems the problem is the missing rundec wheels for Python 3.13. I have opened PR https://github.com/DavidMStraub/rundec-python/pull/16 to fix this.

**Reply by Joro20** [reply]
Should I then wait for the problem to be fixed ?

**Reply by peterstangl** [reply, marked as answer]
> Should I then wait for the problem to be fixed ?

Everything should work on Python 3.6 - 3.12. So if you don't want to wait you can use any of those Python versions. If you want to use Python 3.13, you will have to wait until `rundec-python` (one of the dependencies of `wilson`) supports Python 3.13 (which was just released two weeks ago).

**Reply by Joro20** [reply]
Thank you for explaining the reason. I've been able to use Flavio with Python version 3.12 and now my main issue is also solved.
